src/console/Program.cs

```diff
@@ -27,47 +30,95 @@ public class Program
         public static async Task Main()
         {
             var config = GetEnvironmentConfig();
+            var eventHubName = GetEventHubName(config);
 
-            // determine which event hub to read from
-            var eventHub = Environment.GetEnvironmentVariable("WEBJOBS_NAME");
-            if (eventHub == null)
-            {
-                eventHub = config.GetSection("Console:EventHub").Value;
-            }
-
-            System.Console.WriteLine($"Reading from event hub: {eventHub}");
-            System.Console.WriteLine($"Logs and Metrics will be written to Application Insights");
-            var eventHubOptions = GetEventHubInfo(config, eventHub);
+            var eventHubOptions = GetEventHubInfo(config, eventHubName);
 
             EnsureArg.IsNotNullOrWhiteSpace(eventHubOptions.EventHubConnectionString);
             EnsureArg.IsNotNullOrWhiteSpace(eventHubOptions.EventHubName);
 
-            var eventBatchingOptions = new EventBatchingOptions();
-            config.GetSection(EventBatchingOptions.Settings).Bind(eventBatchingOptions);
-
-            var serviceProvider = GetRequiredServiceProvider(config, eventHub);
+            var serviceProvider = GetRequiredServiceProvider(config, eventHubName);
             var logger = serviceProvider.GetRequiredService<ITelemetryLogger>();
-            var eventConsumers = GetEventConsumers(config, eventHub, serviceProvider, logger);
+
+            var eventConsumers = GetEventConsumers(config, eventHubName, serviceProvider, logger);
 
             var storageOptions = new StorageCheckpointOptions();
             config.GetSection(StorageCheckpointOptions.Settings).Bind(storageOptions);
-            storageOptions.BlobPrefix = eventHub;
-            var checkpointClient = new StorageCheckpointClient(storageOptions, logger);
 
+            var storageCheckpointClient = GetStorageCheckpointClient(storageOptions, logger, eventHubName);
             var eventConsumerService = new EventConsumerService(eventConsumers, logger);
 
+            var incomingEventReader = GetEventProcessorClient(storageCheckpointClient.GetBlobContainerClient(), eventHubOptions);
+            var eventHubReader = GetEventProcessor(config, eventConsumerService, storageCheckpointClient, logger);
+
+            System.Console.WriteLine($"Reading from event hub: {eventHubName}");
+            System.Console.WriteLine($"Logs and Metrics will be written to Application Insights");
             var ct = new CancellationToken();
+            await eventHubReader.RunAsync(incomingEventReader, ct);
+        }
+
+        public static string GetEventHubName(IConfiguration config)
+        {
+            var eventHub = Environment.GetEnvironmentVariable("WEBJOBS_NAME");
```

Comment:
>Environment.GetEnvironmentVariable("WEBJOBS_NAME"); [](start = 27, length = 51)

Where does the WEBJOBS_NAME environment variable get set?  I would expect it to be a property on the web config in the Web Template but I don't see it.

Reply:
Azure Web Jobs sets WEBJOBS_NAME automatically when a web job is created/running. The reason I used this rather than the app service environment variables is that we need to be running two web jobs (normalization, fhir conversion) on the same app service and needed some way for the web job to figure out what it should be doing. I'm open to better ways of doing this.

Reply:
Okay, I am still not following.  I will try to dig in a little more to figure it out.  I may sync with you to understand better.

---
In reply to: [567487431](https://github.com/microsoft/iomt-fhir/pull/87#discussion_r567487431) [](ancestors = 567487431)